src/amplitude-client.js

```diff
@@ -39,7 +39,13 @@ var AmplitudeClient = function AmplitudeClient(instanceName) {
   this._instanceName = utils.isEmptyString(instanceName) ? Constants.DEFAULT_INSTANCE : instanceName.toLowerCase();
   this._unsentEvents = [];
   this._unsentIdentifys = [];
-  this.options = { ...DEFAULT_OPTIONS, trackingOptions: { ...DEFAULT_OPTIONS.trackingOptions } };
+  this.options = {
+    ...DEFAULT_OPTIONS,
```

Comment:
`...` operation only creates a clone for the first-level key values, but reuses the references of the nested ones like `headers`, `library`, `plan` and ` trackingOptions`. In this case, we need a deep clone vs shallow clone.